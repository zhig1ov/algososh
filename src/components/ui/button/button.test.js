import React from "react"
import renderer from "react-test-renderer"
import { render, screen, fireEvent } from "@testing-library/react"
import { Button } from "./button"

describe('Работа компонента button', () => {
	it('Кнопка с текстом рендерится без ошибок', () => {
    const tree = renderer
			.create(<Button text='Нажми меня!'/>)
			.toJSON()
		expect(tree).toMatchSnapshot()            
    })

		it('Кнопка без текста рендерится без ошибок', () => {
			const tree = renderer
				.create(<Button />)
				.toJSON()
			expect(tree).toMatchSnapshot()
		})

		it('Заблокированная кнопка рендерится без ошибок', () => {
			const tree = renderer
				.create(<Button disabled={true} />)
				.toJSON()
			expect(tree).toMatchSnapshot()
		})
		
		it('Кнопка с индикацией загрузки рендерится без ошибок', () => {
			const tree = renderer	
				.create(<Button isLoader={true} />)
				.toJSON()
			expect(tree).toMatchSnapshot()
		})

		it('Коллбэк кнопки работает корректно', () => {
			window.alert = jest.fn()
			render(<Button text='Жмякни меня!' onClick={alert('Booms')} />)
			const button = screen.getByText('Жмякни меня!')
			fireEvent.click(button)
			expect(window.alert).toHaveBeenCalledWith('Booms')
		})
})