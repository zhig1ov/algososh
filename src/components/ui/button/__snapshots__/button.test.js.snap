// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Работа компонента - button Заблокированная кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Работа компонента - button Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Работа компонента - button Кнопка с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Работа компонента - button Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня!
  </p>
</button>
`;

exports[`Работа компонента button Заблокированная кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Работа компонента button Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Работа компонента button Кнопка с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Работа компонента button Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня!
  </p>
</button>
`;
